src/main/java/org/xstefank/model/PersistentList.java

```diff
@@ -30,32 +33,146 @@ public PersistentList(String dirName, String fileName) {
         this(new File(dirName, fileName));
     }
 
-    public void addUser(String username) {
-        if (list.contains(username)) {
+    @Override
+    public String set(int index, String element) {
+        String oldValue = super.set(index, element);
+        rewriteFileContent(index);
+        return oldValue;
+    }
+
+    @Override
+    public boolean add(String s) {
+        if (contains(s) || !super.add(s)) {
+            return false;
+        }
+        try (FileWriter fw = new FileWriter(file, true)) {
+            fw.write(s + System.lineSeparator());
```

Comment:
I agree up to a point.. If I call method `super.addAll(Collection)` it would not work because I will have to add line separator next to every single element in that collection.

Reply:
@lukasburda sorry I don't follow... How does that relate to extracting lineseparator to a field as I was saying in the original comment? You're code will be the same except for 
```suggestion
            fw.write(s + LINE_SEPARATOR);
```

Reply:
Sorry I misunderstood.